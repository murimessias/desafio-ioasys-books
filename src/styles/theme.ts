export default {
  border: {
    radius: {
      md: '0.4rem',
      rounded: '9999px',
    },
  },

  colors: {
    white: '#FFFFFF',
    caramel: '#F5F5EE',
    black: '#333333',
    gray: '#999999',
    accent: '#AB2680',
  },

  font: {
    family:
      "Heebo, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",

    sizes: {
      sm: '1.2rem',
      md: '1.4rem',
      lg: '1.6rem',
      xl: '2.8rem',
    },

    weight: {
      normal: 400,
      bold: 500,
    },
  },

  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50,
  },

  leading: {
    none: 1,
    tight: 1.25,
    normal: 1.5,
  },

  spacings: {
    xs: '1.6rem',
    sm: '2.4rem',
    md: '3.2rem',
    lg: '4.0rem',
    xl: '4.8rem',
  },
} as const
