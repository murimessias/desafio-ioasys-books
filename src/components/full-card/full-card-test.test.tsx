import { screen } from '@testing-library/react'
import { renderWithTheme } from 'resources/utils/tests/helpers'
import theme from 'styles/theme'

import { FullCard } from '.'

const props = {
  imageUrl: '/img/full-book-img-placeholder.png',
  title: 'Change By Design Second line exa',
  authors: ['Nir Eyal', 'Ryan Hoover', 'John Dude', 'Carls Marg', 'Adam Paul'],
  pageCount: 150,
  publisher: 'Loyola',
  published: 2020,
  language: 'Inglês',
  isbn10: '0062856626',
  isbn13: '978-0062856623',
  description:
    'The subject of “design thinking” is the rage at business schools, throughout corporations, and increasingly in the popular press—due in large part to the work of IDEO, a leading design firm, and its celebrated CEO, Tim Brown, who uses this book to show how the techniques and strategies of design belong at every level of business.',
}

describe('<FullCard />', () => {
  it('should render the full card correctly', () => {
    const { container } = renderWithTheme(<FullCard {...props} />)

    expect(
      screen.getByRole('heading', {
        name: /change by design second line exa/i,
      }),
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the data correctly', () => {
    renderWithTheme(<FullCard {...props} />)

    expect(screen.getByRole('img')).toBeInTheDocument()

    const fontSize = theme.font.sizes.xl
    expect(
      screen.getByRole('heading', {
        name: /change by design second line exa/i,
      }),
    ).toHaveStyle({
      'font-size': fontSize,
    })
  })
})
