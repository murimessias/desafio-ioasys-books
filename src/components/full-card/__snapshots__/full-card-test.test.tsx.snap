// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullCard /> should render the full card correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFFFFF;
  border-radius: 0.4rem;
  box-shadow: 0 0.6rem 2.4rem rgba(84,16,95,0.15);
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 2.4rem;
  -webkit-transition: all 250ms ease-in-out;
  transition: all 250ms ease-in-out;
  width: 100%;
}

.c0 h4 {
  color: #333333;
  margin-bottom: 1.6rem;
  text-transform: uppercase;
}

.c0 div + div {
  margin-top: 2.4rem;
}

.c1 {
  display: block;
  height: 100%;
  position: relative;
}

.c1 img {
  box-shadow: 0 0.6rem 2.4rem rgba(84,16,95,0.45);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  margin-bottom: 0.8rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c6 {
  color: #AB2680;
  font-weight: 400;
  list-style: none;
  padding-right: 0.4rem;
}

.c4 {
  color: #333333;
  font-size: 2.8rem;
  line-height: 1.25;
  -webkit-letter-spacing: -0.025rem;
  -moz-letter-spacing: -0.025rem;
  -ms-letter-spacing: -0.025rem;
  letter-spacing: -0.025rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 0.4rem;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c8 span:first-child {
  color: #333333;
  font-weight: 500;
}

.c9 {
  height: 100%;
}

.c9::-webkit-scrollbar {
  width: 0.2rem;
}

.c9::-webkit-scrollbar-track {
  background: transparent;
}

.c9::-webkit-scrollbar-thumb {
  background: #AB2680;
  border-radius: 1rem;
}

.c9::-webkit-scrollbar-thumb:hover {
  background: #AB2680;
  -webkit-filter: contrast(0.9);
  filter: contrast(0.9);
}

.c9::before {
  bottom: 0.2rem;
  content: url('/icons/quotes.svg');
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 1.6rem;
  vertical-align: middle;
  margin-right: 0.4rem;
  position: relative;
  width: 1.4rem;
}

.c9 p {
  display: inline;
}

@media (min-width:768px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:768px) {
  .c1 {
    margin-right: 1.6rem;
  }
}

@media (min-width:768px) {
  .c9 {
    height: 8rem;
    overflow-y: scroll;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="Change By Design Second line exa"
      layout="fill"
      src="/img/full-book-img-placeholder.png"
    />
  </div>
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <h2
        class="c4"
      >
        Change By Design Second line exa
      </h2>
      <ul
        class="c5"
      >
        <li
          class="c6"
        >
          Nir Eyal
          ,
        </li>
        <li
          class="c6"
        >
          Ryan Hoover
          ,
        </li>
        <li
          class="c6"
        >
          John Dude
          ,
        </li>
        <li
          class="c6"
        >
          Carls Marg
          ,
        </li>
        <li
          class="c6"
        >
          Adam Paul
        </li>
      </ul>
    </div>
    <div
      class="c7"
    >
      <h4>
        Informações
      </h4>
      <ul>
        <li
          class="c8"
        >
          <span>
            Páginas
          </span>
          <span>
            150
             páginas
          </span>
        </li>
        <li
          class="c8"
        >
          <span>
            Editora
          </span>
          <span>
            Editora 
            Loyola
          </span>
        </li>
        <li
          class="c8"
        >
          <span>
            Publicação
          </span>
          <span>
            2020
          </span>
        </li>
        <li
          class="c8"
        >
          <span>
            Idioma
          </span>
          <span>
            Inglês
          </span>
        </li>
        <li
          class="c8"
        >
          <span>
            ISBN-10
          </span>
          <span>
            0062856626
          </span>
        </li>
        <li
          class="c8"
        >
          <span>
            ISBN-13
          </span>
          <span>
            978-0062856623
          </span>
        </li>
      </ul>
    </div>
    <div
      class=""
    >
      <h4>
        Resenha da Editora
      </h4>
      <blockquote
        class="c9"
      >
        <p>
          The subject of “design thinking” is the rage at business schools, throughout corporations, and increasingly in the popular press—due in large part to the work of IDEO, a leading design firm, and its celebrated CEO, Tim Brown, who uses this book to show how the techniques and strategies of design belong at every level of business.
        </p>
      </blockquote>
    </div>
  </div>
</div>
`
